From c2f193cc422eac475bbc17261413c83ff13172d4 Mon Sep 17 00:00:00 2001
From: Alexander Tsoy <alexander@tsoy.me>
Date: Mon, 25 May 2020 17:49:20 +0300
Subject: [PATCH] busybox: simplify listing of supported utilities

'--list' option is supported since busybox-1.20.0, which was released
in 2010.

(cherry picked from commit 50cc23ba32b0fda63eff7623b529dbeb4e6a38c6)

Resolves: #1959336
---
 modules.d/05busybox/module-setup.sh | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/modules.d/05busybox/module-setup.sh b/modules.d/05busybox/module-setup.sh
index ecbd6a13..5d88c5d1 100755
--- a/modules.d/05busybox/module-setup.sh
+++ b/modules.d/05busybox/module-setup.sh
@@ -14,15 +14,16 @@ depends() {
 
 # called by dracut
 install() {
-    local _i _progs _path _busybox
+    local _i _path _busybox
+    local _progs=()
     _busybox=$(type -P busybox)
     inst $_busybox /usr/bin/busybox
-    for _i in $($_busybox | sed -ne '1,/Currently/!{s/,//g; s/busybox//g; p}')
-    do
-        _progs="$_progs $_i"
+    for _i in $($_busybox --list); do
+        [[ ${_i} == busybox ]] && continue
+        _progs+=("${_i}")
     done
 
-    for _i in $_progs; do
+    for _i in "${_progs[@]}"; do
         _path=$(find_binary "$_i")
         [ -z "$_path" ] && continue
         ln_r /usr/bin/busybox $_path

